### ISSUE TITLE
Numbers inserted in the wrong order

### ISSUE THREAD

**Issue by axefrog** [zdivozzo/increment-me#2]
The package seems to have stopped inserting numbers in the correct order. An animation is worth a thousand words:

![atom-zdivozzo-increment-me-wrong-order](https://cloud.githubusercontent.com/assets/298883/14136719/8f03711c-f6a7-11e5-97fc-931b66655599.gif)

**Comment by zdivozzo** [comment]
Whoa, weird.  I can't replicate the issue, are you using 1.6.1?

Did you install any other packages recently that may be related? Or anything that you have installed related to multiple cursors? Just trying to narrow it down a bit since it seems to be working fine for me.

Just realized I never tagged you, oops  @axefrog

**Comment by potto007** [comment]
This is happening to me, too. I am using 1.8.0 on OS X.

**Comment by morganbye** [comment]
Ditto to v0.3.0 on OS X

A list of 16-70 goes to:
`124
98
106
78
110
120
95
117
112
84
71
76
91
111
82
103
73
100
108
80
118
113
85
92
107
102
115
87
72
121
122
96
83
105
74
94
119
109
81
97
90
77
88
101
123
116
89
75
93
79
114
86
104
99
70`

**Comment by zdivozzo** [comment]
@potto007 @morganbye Sorry I've been AWOL, I've tried this a bunch of times and I can't seem to replicate it.  I obviously believe that it's happening, I just don't know why.  I'm guessing its getting interfered with by another package, can one of you try just running it from the command palette and see if it does the same thing?  Do you have anything else installed related to multiple cursors that I can install and see if it has this behavior?
Thanks!

**Comment by fabianfiorotto** [comment]
I disabled all packages and it still happens. OS Linux mint 17 atom v1.8.9

**Comment by morganbye** [comment]
Yeah, likewise, seems to be package independent and doesn't matter if I run from the keyboard shortcut or command palette.

Perhaps, most interesting if I go to undo the operation (repeatedly hitting Cmd/Ctrl-Z) then the cursor jumps up and down randomly to undo the operation, one number at a time, but the undo is done systematically. i.e. it undoes the highest number first

**Comment by potto007** [comment]
I've been able to consistently recreate this behavior. I decided to step-debug this, and found the selections array populated by editor.getSelections() inside convert held cursors in the same order I eventually saw the numbers to be in. So, looking at the Atom codebase: https://github.com/atom/atom/blob/v1.9.9/src/text-editor.coffee#L2490-L2502
getSelections: Get current {Selection}s
getSelectionsOrderedByBufferPosition: Get all {Selection}s, ordered by their position in the buffer instead of the order in which they were added.
The second function implies that the first function assembles the selection array in the order those selections were added to the Buffer, not the order in which they appear in the Buffer.
